### PR TITLE
Add tslint

### PR DESCRIPTION
**Pull request by Katharine**
This PR:

- Adds a functional `hack/verify-tslint.sh`
- Makes all typescript pass the linter
- Adds a tslint job for test-infra.

I'd recommend reviewing the commits individually.

### CODE REVIEW

File: config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml [hunk 1]
```diff
@@ -94,6 +94,18 @@ presubmits:
           requests:
             memory: "2Gi"
 
+  - name: pull-test-infra-verify-tslint
+    branches:
+      - master
+    decorate: true
```
Comment (krzyzacy) [inline config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml]: add a run_if_changed against all `*.ts`?

File: tslint.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        "interface-name": false,
+        "quotemark": false,
+        "no-console": false,
+        "max-line-length": false,
+        "indent": [true, "spaces", 2],
+        "max-classes-per-file": false,
+        "prefer-template": [true, "allow-single-concat"]
+    },
+    "rulesDirectory": []
+}
```
Comment (krzyzacy) [inline tslint.json]: nit: new line

File: config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml (hunk 1 above)
Comment (Katharine) [reply config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml]: Done.

File: tslint.json (hunk 2 above)
Comment (Katharine) [reply tslint.json]: Fixed.

**Pull request merged**